### PR TITLE
Add tenureType call 

### PR DESCRIPTION
**Pull request by siapankina**
### Description of change

Add Tenure type
Add test to check that response is correct when tenancies is an empty array

### Story Link
https://trello.com/c/JeorEmlt/154-as-a-user-i-need-to-see-tenure-type-so-i-know-to-raise-a-repair

### CODE REVIEW

File: app/models/tenure.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  attr_accessor :typeCode, :typeDescription
+
+  class << self
+    def find_tenure_type(reference)
+      response = PlatformApis::TenancyInformation::Client.get_tenancy_information_by_property_reference(reference)
+
+      build(response["tenancies"].first["tenureType"])
+    end
+
+    def build(attribute)
```
Comment (neilvanbeinum) [inline app/models/tenure.rb:21]: Please could we add a small comment with an example of the tenureType value, to see why we're splitting things.

File: app/models/tenure.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  attr_accessor :typeCode, :typeDescription
+
+  class << self
+    def find_tenure_type(reference)
+      response = PlatformApis::TenancyInformation::Client.get_tenancy_information_by_property_reference(reference)
+
+      # The API returns a 200 with response "tenancies": [] when no
+      # tenancy agreement reference can be found for a property.
+      # In this case we will return nil if response["tenancies"] is empty
+      build(response["tenancies"].first["tenureType"]) if response["tenancies"].any?
```
Comment (benbaumann95) [inline app/models/tenure.rb:15]: Hey If I use prop ref `00072708` for example then this would return `"tenure": null` on the API. This might be what we want but just putting this here in case we wanted to return something else

**Review by benbaumann95 — APPROVED** [review]
Hey this is looking good, great job 👍 
Just a quick question before we move forward, is `typeCode` and `typeDescription` HACT-compliant naming ?

File: spec/requests/properties_spec.rb [hunk 3]
```diff
@@ -187,123 +187,194 @@
...
-            {
-              propRef: "100023022310",
-              address1: "1 Example Road",
-              postCode: "A1 1AA",
-              levelCode: "1",
-              subtypCode: "DWE"
-            },
-          status: 200
-        )
+      context "something for now" do
```
Comment (benbaumann95) [inline spec/requests/properties_spec.rb]: Hey was this meant to be changed? 😄

Comment (benbaumann95) [reply spec/requests/properties_spec.rb:190]: Do we need another context block here?

File: spec/requests/properties_spec.rb [hunk 4]
```diff
@@ -187,123 +187,194 @@
...
-              "levelCode" => "1",
-              "subTypeCode" => "DWE",
-              "subTypeDescription" => "Dwelling"
-            }
-          },
-          "alerts" => {
-            "locationAlert" => [
+      context "with no tenancy information details" do
+        before do
+          stub_property_request(
```
Comment (benbaumann95) [inline spec/requests/properties_spec.rb]: You could probably put this in the block above if it's the same as the above `stub_property_request` https://github.com/LBHackney-IT/repairs-api/pull/21/files#diff-2fe1122181c84f9dbd72fcc3d2629c12530a2a15848dcc3ffc788460f2121f6aR192 to avoid repeating yourself

Comment (benbaumann95) [reply spec/requests/properties_spec.rb]: Higher level context block I mean

File: spec/requests/properties_spec.rb [hunk 5]
```diff
@@ -187,123 +187,194 @@
...
-                "endDate" => nil
+                "tenancies": []
+              },
+            status: 200
+          )
+
+          get("/api/v2/properties/100012345678", headers: headers)
+        end
+
+        it "returns a JSON representation from a single property with tenure=nil and personAlert=[]" do
```
Comment (benbaumann95) [inline spec/requests/properties_spec.rb]: I suppose `locationAlert=[]` as well so you could just say no alerts . It's probably better to have tests as human readable so I would also say no tenure details instead of `tenure=nil` too

File: spec/requests/properties_spec.rb (hunk 3 above)
Comment (siapankina) [reply spec/requests/properties_spec.rb]: oh, sorry ! yes, need to change it :D

File: app/models/tenure.rb (hunk 2 above)
Comment (siapankina) [reply app/models/tenure.rb:15]: yup, that's what we want
will update the comment

**Review by EGiataganas — APPROVED** [review]
